src/library-extender/README.md

```diff
@@ -0,0 +1,93 @@
+# Custom Digital Assets Management provider
+
+> IMPORTANT - Please note that there is a known limitation, you cannot use the Angular binding syntax {{item.data.CreatedBy}}, when creating component HTML templates, you must use another binding, for example [textContent]="item.data.CreatedBy", or [innerHtml]="item.data.Content".
+
+You can add custom DAM providers if the one you are using is not supported by default from Sitefinity - Frontify and Cloudinary.
+    
+    Note: There must be only one enabled DAM provider on the server - 
+          In your case ths should be the server-side implementation of your custom DAM provider.
+
+In order to register new provider you need to create new class which extends **DamProviderBase**.
+```typescript
+import { ClassProvider, Injectable } from "@angular/core";
+import { DamProviderBase, DAM_PROVIDER_TOKEN, EntityData } from "@progress/sitefinity-adminapp-sdk/app/api/v1";
+import { DamAsset } from "@progress/sitefinity-adminapp-sdk/app/api/v1/dam/dam-asset";
+
+declare var cloudinary: any;
+
+const CUSTOM_MEDIA_CANNOT_BE_LOADED = "Custom media selector cannot be loaded.";
+const CUSTOM_PROVIDER_TYPE_NAME = "CloudinaryBlobStorageProvider";
+
+@Injectable()
+class CustomDamProvider extends DamProviderBase {
+    isSupported(providerTypeName: string): boolean {
+        ...
+    }
+    
+    loadMediaSelector(damWrapper: HTMLElement, mediaEntityData: EntityData, allowMultiSelect: boolean): void {
+        ...
+    }
+}    
+```
+There are two abstract methods that need to be implemented because they are specific for each provider:
+* isSupported - Use this method to determine if your provider supports the one that is enabled on the server. As parameter it receives the name of the provider which is defined server-side.
```

Comment:
"it receives the name of the provider which is defined" -> It receives the name of the Provider Type. You can name the provider anyway you want, what matters is the name of the Class on the server.